### ISSUE TITLE
Add explicit deprecation warning for flax.nn

### ISSUE THREAD

**Issue by avital** [google/flax#906]
`flax.nn` is planned to be officially no longer supported during May 2021 (in favor of `flax.linen`). Let's start putting deprecation warnings now, so that people aren't surprised too much later. /cc @marcvanzee

**Comment by avital** [comment]
Also good to add deprecation messages to all docstrings. I just searched for "Flax Batch Norm" and found this: https://flax.readthedocs.io/en/latest/_autosummary/flax.nn.BatchNorm.html

A lot of users won't notice the small "(deprecated)" at the top in the nav bar

**Comment by joaogui1** [comment]
Opened a PR fixing this @avital

**Comment by avital** [comment]
go for it @joaogui1!

**Comment by joaogui1** [comment]
I think this can be closed :D

**Comment by avital** [comment]
Yup! Closed by #1001
